### PR TITLE
Mobile library search bar

### PR DESCRIPTION
**Pull request by axel-denis**
Fixes ARE-381

### CODE REVIEW

File: frontend/mobile/lib/pages/library.dart [hunk 1]
```diff
@@ -32,34 +35,75 @@ class _LibraryState extends State<Library> {
...
+          child: Column(
+            children: [
+              Padding(
+                padding: const EdgeInsets.all(4.0),
+                child: Container(
+                  color: Colors.white,
+                  child: TextField(
+                    controller: searchController,
+                    decoration: InputDecoration(
+                      hintText: 'Search...',
```
Comment (RezaRahemtola) [inline frontend/mobile/lib/pages/library.dart:47]: add this in translation files

File: frontend/mobile/lib/pages/library.dart [hunk 2]
```diff
@@ -32,34 +35,75 @@ class _LibraryState extends State<Library> {
...
+                      suffixIcon: IconButton(
+                        icon: const Icon(Icons.clear),
+                        onPressed: () => searchController.clear(),
+                      ),
+                      prefixIcon: IconButton(
+                        icon: const Icon(Icons.search),
+                        onPressed: () async {
+                          setState(() {
+                            workflows = initialWorkflows
```
[the remark this answers is not in the document]
Comment (axel-denis) [reply frontend/mobile/lib/pages/library.dart:56]: No, I needed to separate the search results from the initial results, to allow the user to make many research without destroying initial data

**Review by Tomi-Tom — APPROVED** [review]

File: frontend/mobile/lib/pages/library.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (RezaRahemtola) [reply frontend/mobile/lib/pages/library.dart:56]: oh ok

File: frontend/mobile/lib/pages/library.dart [hunk 3]
```diff
@@ -14,15 +14,18 @@ class Library extends StatefulWidget {
 
 class _LibraryState extends State<Library> {
   late List<Workflow> workflows;
-
   @override
   void initState() {
     super.initState();
     workflows = [];
   }
```
Comment (RezaRahemtola) [inline frontend/mobile/lib/pages/library.dart:21]:
```suggestion
  @override
  void initState() {
    super.initState();
  }
```

File: frontend/mobile/lib/pages/library.dart [hunk 4]
```diff
@@ -14,15 +14,18 @@ class Library extends StatefulWidget {
 
 class _LibraryState extends State<Library> {
   late List<Workflow> workflows;
```
Comment (RezaRahemtola) [inline frontend/mobile/lib/pages/library.dart:16]:
```suggestion
  List<Workflow> workflows = [];
```

File: frontend/mobile/lib/pages/library.dart [hunk 5]
```diff
@@ -14,15 +14,18 @@ class Library extends StatefulWidget {
...
-
   @override
   void initState() {
     super.initState();
     workflows = [];
   }
 
+  bool needRefresh = true;
+  List<Workflow> initialWorkflows = [];
```
Comment (RezaRahemtola) [inline frontend/mobile/lib/pages/library.dart:24]: put this before the initState to keep all the variables together

File: frontend/mobile/lib/pages/library.dart [hunk 6]
```diff
@@ -32,34 +35,77 @@ class _LibraryState extends State<Library> {
...
+                      return ListView.builder(
+                        itemCount: workflows.length,
+                        itemBuilder: (context, index) {
+                          return WorkflowTile(
+                            workflow: workflows[index],
+                            onUpdate: () async {
+                              final newWorkflows =
+                                  await services.workflows.getAll();
+                              setState(() {
+                                workflows = newWorkflows.data!;
```
Comment (RezaRahemtola) [inline frontend/mobile/lib/pages/library.dart:98]: apply the search here too, or when somebody renames a workflow it won't be applied

Comment (RezaRahemtola) [reply frontend/mobile/lib/pages/library.dart:98]: or you can also clear the search, but the first solution is better imo

**Review by RezaRahemtola — CHANGES_REQUESTED** [review]

File: frontend/mobile/lib/pages/library.dart (hunk 6 above)
Comment (axel-denis) [reply frontend/mobile/lib/pages/library.dart:97]: this is only ran once at the page loading, so no need to apply the search here

Comment (RezaRahemtola) [reply frontend/mobile/lib/pages/library.dart:98]: it's also ran when somebody updates the workflow (rename or delete for example).
So here I have that if I have 5 workflows, search for "unknown" (that marches a single workflow), if I delete or rename it then all the other workflows will appear again

Comment (RezaRahemtola) [reply frontend/mobile/lib/pages/library.dart]: fixed

**Review by RezaRahemtola — APPROVED** [review]

**Pull request merged**